File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        /**
        [100,4,200,1,3,2]
                   ^
          200-----
        */
        Set<Integer> set = new HashSet<>();
        for (int n : nums) set.add(n);
        int longest = 0;
        for (int n : set) {
            if (set.contains(n - 1)) continue;
            int streak = 0;
            while (set.contains(n)) {
                streak++;
                n++;
            }
            longest = Math.max(longest, streak);

        }
        return longest;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        longest = 0
        for n in nums_set:
            if n - 1 in nums_set:
                continue
            curr = 0
            i = n
            while i in nums_set:
                curr += 1
                i += 1
            longest = max(longest, curr)

        return longest